### PR TITLE
Cherry-pick #16988 to 7.x: New input for CrowdStrike Falcon events

### PR DESCRIPTION
**Pull request by tonymeehan**
Cherry-pick of PR elastic/beats#16988 to 7.x branch. Original message:

## What does this PR do?

This adds a new input for CrowdStrike Falcon events forwarded by CrowdStrike's [SIEM forwarder](https://www.crowdstrike.com/blog/tech-center/integrate-with-your-siem/). This input uses the default JSON output format from the SIEM forwarder.

## Why is it important?

We've had several users asks us for adding support for CrowdStrike Falcon in ECS. We strive to be data agnostic to help enable our SOC users aggregate all of their data in our SIEM. Using some log data provided by our users, I've put this new module together to ingest the data and convert many fields into ECS (including categorization).

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have made corresponding change to the default configuration files
- [X] I have added tests that prove my fix is effective or that my feature works

## How to test this PR locally

```
mage update build integTest
```

## Screenshots

crowdstrike mov

## Proof this works

```
[success] 0.85% test_xpack_modules.XPackTest.test_fileset_file_008_crowdstrike: 2.7708s
----------------------------------------------------------------------
Ran 135 tests in 326.929s

OK
```

I also ran ```make docs``` to validate the documentation looked good.

## Other comments

I'll squash before merging.

### CODE REVIEW

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -318,6 +318,8 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 - Add Filebeat Okta module. {pull}16362[16362]
 - Add source field in k8s events {pull}17209[17209]
 - Improve AWS cloudtrail field mappings {issue}16086[16086] {issue}16110[16110] {pull}17155[17155]
+- Added new module `crowdstrike` for ingesting Crowdstrike Falcon streaming API endpoint event data. {pull}16988[16988]
+- Added documentation for running Filebeat in Cloud Foundry. {pull}17275[17275]
```
Comment (leehinman) [inline CHANGELOG.next.asciidoc:322]: just want the Changelog line for your PR.  (this happens with backports)

Comment (tonymeehan) [reply CHANGELOG.next.asciidoc:322]: Ah I missed that. Thank you!

File: CHANGELOG.next.asciidoc [hunk 2]
```diff
@@ -318,10 +318,11 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
...
 - Move azure-eventhub input to GA. {issue}15671[15671] {pull}17313[17313]
 - Improve ECS categorization field mappings in mongodb module. {issue}16170[16170] {pull}17371[17371]
 - Improve ECS categorization field mappings for mssql module. {issue}16171[16171] {pull}17376[17376]
-- Added documentation for running Filebeat in Cloud Foundry. {pull}17275[17275]
```
Comment (leehinman) [inline CHANGELOG.next.asciidoc:324]: looks like you removed the wrong "pull 17275" line.  You want to remove the one next to your Crowdstrike line.

**Comment by tonymeehan** [comment]
@leehinman should I update this PR with the fix from https://github.com/elastic/beats/pull/17819 so that tests pass here? Looks like the failing test is due to the issue identified in #17819.

**Pull request merged**